elasticai/creator/vhdl/language.py

```diff
@@ -99,6 +103,92 @@ def __init__(self, identifier: str):
         super().__init__(identifier, Keywords.COMPONENT)
 
 
+class Architecture:
+    def __init__(self, identifier: str, design_unit: str, process_content: str):
+        self.identifier = identifier
+        self.design_unit = design_unit
+        self.process_content = process_content
+
+    def _header(self) -> Code:
+        yield self.process_content
+
+    def __call__(self) -> Code:
+        return _wrap_in_ARCHITECTURE_block(
+            block_type=Keywords.ARCHITECTURE,
+            block_identifier=self.identifier,
+            lines=self._header(),
+            design_unit_identifier=self.design_unit,
+        )
+
+
+class Process:
+    def __init__(
+        self, identifier: str, input: str, lookup_table_generator_function: str
+    ):
+        self.identifier = identifier
+        self._variable_initialization_list = InterfaceList()
+        # FIXME: change name
+        self._variables_list = InterfaceList()
+        self.lookup_table_generator_function = lookup_table_generator_function
+        self.input = input
+
+    @property
+    def variable_initialization_list(self):
+        return self._variable_initialization_list
+
+    @variable_initialization_list.setter
+    def variable_initialization_list(self, value):
+        self._variable_initialization_list = InterfaceList(value)
+
+    @property
+    def variable_list(self):
+        return self._variables_list
+
+    @variable_list.setter
+    def variable_list(self, value):
+        self._variables_list = InterfaceList(value)
+
+    def _header(self) -> Code:
+        if len(self.variable_initialization_list) > 0:
+            yield from _add_semicolon_in_each_line(self._variable_initialization_list())
+
+    def _footer(self) -> Code:
+        if len(self.variable_list) > 0:
+            yield from _add_semicolon_in_each_line(self._variables_list())
+        yield self.lookup_table_generator_function
+
+    def __call__(self) -> Code:
+        return _wrap_in_PROCESS_block(
+            block_type=Keywords.PROCESS,
+            block_identifier=self.identifier,
+            input=self.input,
+            lines_header=self._header(),
+            lines_footer=self._footer(),
+        )
+
+
+class Library:
+    def __init__(self):
+        self._more_libraries_list = InterfaceList()
```

Comment:
I think we shouldn't be treating the first three libs different from the `_more_libraries_list`?

Reply:
I'd suggest we do something like:
```python
class Library:
  _library_header = "library ieee;" # no idea if this is actually the library header, better look it up in the reference
  def __init__(self):
    self._std_libs = ["ieee.std_logic_1164.all", "ieee.numeric_std.all"]
    self._more_libs = []

  @property
  def libs(self):
    yield from chain(self._std_libs, self._more_libs)

  def __call__(self):
    yield self._library_header
    yield from _add_semicolon_in_each_line(self.prefix_lines_with_USE(self.libs))
```

Reply:
In addition i'm wondering: Are these InterfaceLists as well? If they are actually a subtype of InterfaceList, that behaves/is built exactly the same we could alias it:
```
IdenticalToInterfacesList = InterfaceList
```
Otherwise we could actually derive from InterfaceList
```
class IdenticalToInterfaceList(InterfaceList):
  pass
```

Reply:
ah okay, I didn't get in the first draft that the interface list refers to an interface of vhdl. I just thought it is an interface of a list. I changed it to a normal list like you suggested

Reply:
I'm just not sure about the prefix_use function because the lstm_cell.vhd has something like `LIBRARY work;` without use in before

Reply:
hmm.... i guess a library declaration would then look something like this:
```
library <lib>;
use <lib>.<package>;
use <lib>.<other_package>;
```

So i guess in the lstm file we're seeing two libraries, whereas we got only one here. With this in mind i don't think my proposed code makes sense.